pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1355,73 +1355,42 @@ func doInactivate(ctx *domainContext, status *types.DomainStatus, impatient bool
 		status.State = types.HALTING
 		publishDomainStatus(ctx, status)
 
-		switch status.VirtualizationMode {
-		case types.HVM, types.FML:
-			// Do a short shutdown wait, just in case there are
-			// PV tools in guest, then a shutdown -F
-			// If the Shutdown fails we don't wait; assume failure
-			// was due to no PV tools
-			shortDelay := time.Second * 60
-			if err := DomainShutdown(*status, false); err != nil {
-				log.Errorf("DomainShutdown %s failed: %s",
-					status.DomainName, err)
-			} else {
-				// Wait for the domain to go away
-				log.Functionf("doInactivate(%v) for %s: waiting for domain to shutdown",
-					status.UUIDandVersion, status.DisplayName)
-				gone := waitForDomainGone(*status, shortDelay)
-				if gone {
-					status.DomainId = 0
-					break
-				}
-			}
-			// This often fails with "X is an invalid domain identifier (rc=-6)"
-			// due to the DomainShutdown above, in which case
-			// the domain is already on the way down.
-			// In that case we wait
-			if err := DomainShutdown(*status, true); err != nil {
-				log.Warnf("DomainShutdown -F %s failed: %s",
-					status.DomainName, err)
-			} else {
-				log.Functionf("doInactivate(%v) for %s Shutdown(force) succeeded",
-					status.UUIDandVersion, status.DisplayName)
-			}
+		// Do a short shutdown wait, just in case there are
+		// PV tools in guest, then a shutdown -F
+		// If the Shutdown fails we don't wait; assume failure
+		// was due to no PV tools
+		shortDelay := time.Second * 60
+		if err := DomainShutdown(*status, false); err != nil {
+			log.Errorf("DomainShutdown %s failed: %s",
+				status.DomainName, err)
+		} else {
 			// Wait for the domain to go away
-			log.Functionf("doInactivate(%v) for %s: waiting for domain to poweroff",
+			log.Functionf("doInactivate(%v) for %s: waiting for domain to shutdown",
 				status.UUIDandVersion, status.DisplayName)
-			gone := waitForDomainGone(*status, maxDelay)
+			gone := waitForDomainGone(*status, shortDelay)
```

Comment:
Here you are accidentally using the shortDelay even in the case of PV; that should use maxDelay.
Might make sense introducing a new shutdownDelay which is set differently based on mode?